conans/test/functional/layout/test_in_cache.py

```diff
@@ -26,7 +26,7 @@ def layout(self):
     def source(self):
         self.output.warn("Source folder: {}".format(self.source_folder))
         # The layout describes where the sources are, not force them to be there
-        tools.save("my_sources/source.h", "foo")
+        tools.save("source.h", "foo")
```

Comment:
This is one evidence IMHO that this change is necessary. Check the ``build()`` ``tools.save()`` without needing to add the subfolder ``my_build`` specified in the ``layout()``. I think it should be symmetric, if ``build()`` executes in the full ``self.build_folder``, including the ``layout`` part, the ``source()`` method should execute in the ``self.source_folder``, including the ``layout`` part.